File: app/src/main/java/miui/statusbar/lyric/hook/app/Kuwo.java
package miui.statusbar.lyric.hook.app;

import android.app.AndroidAppHelper;
import android.content.Context;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.callbacks.XC_LoadPackage;
import miui.statusbar.lyric.utils.Utils;

public class Kuwo {
    static Context context;

    public static class Hook {
        public Hook(XC_LoadPackage.LoadPackageParam lpparam) throws ClassNotFoundException {
            XposedHelpers.findAndHookMethod("android.bluetooth.BluetoothAdapter", lpparam.classLoader, "isEnabled", new XC_MethodHook() {
                @Override
                protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                    super.afterHookedMethod(param);
                    param.setResult(true);
                }
            });
            XposedHelpers.findAndHookMethod("cn.kuwo.mod.playcontrol.RemoteControlLyricMgr", lpparam.classLoader, "updateLyricText", Class.forName("java.lang.String"), new XC_MethodHook() {
                @Override
                protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                    super.beforeHookedMethod(param);
                    String str = (String) param.args[0];
                    Utils.log("酷我音乐:" + str);
                    if (param.args[0] != null && !str.equals("")) {
                        Utils.sendLyric(AndroidAppHelper.currentApplication(), "" + str, "KuWo");
                    }
                    param.setResult(replaceHookedMethod());
                }

                private Object replaceHookedMethod() {
                    return null;
                }
            });
        }
    }
}

File: app/src/main/java/cn/fkj233/xposed/statusbarlyric/hook/app/Kuwo.kt
package cn.fkj233.xposed.statusbarlyric.hook.app

import android.app.AndroidAppHelper
import com.microsoft.appcenter.AppCenter
import com.microsoft.appcenter.analytics.Analytics
import com.microsoft.appcenter.crashes.Crashes
import de.robv.android.xposed.callbacks.XC_LoadPackage
import cn.fkj233.xposed.statusbarlyric.utils.LogUtils
import cn.fkj233.xposed.statusbarlyric.utils.Utils
import cn.fkj233.xposed.statusbarlyric.utils.ktx.hookAfterMethod


class Kuwo(private val lpparam: XC_LoadPackage.LoadPackageParam) {
    fun hook(){
        "android.bluetooth.BluetoothAdapter".hookAfterMethod("isEnabled", classLoader = lpparam.classLoader) {
            it.result = true
        }
        "cn.kuwo.mod.playcontrol.RemoteControlLyricMgr".hookAfterMethod("updateLyricText", String::class.java, classLoader = lpparam.classLoader) {
            AppCenter.start(
                AndroidAppHelper.currentApplication(), "31eda7c3-a1be-4f27-aa45-47d83c513615",
                Analytics::class.java, Crashes::class.java
            )
            val str = it.args[0] as String
            LogUtils.e("酷我音乐:$str")
            if (it.args[0] != null && str != "") {
                Utils.sendLyric(AndroidAppHelper.currentApplication(), "" + str, "KuWo")
            }
            it.result = null
        }
    }
}